src/main/java/frc/robot/subsystems/VisionControl.java

```diff
@@ -74,17 +245,46 @@ private void update() {
         bally = visionData.by;
         ballr = visionData.br;
         wait = visionData.waitForOtherRobot;
+        chassis.updateEncoders();
     }
 
     private void calculateShooter() {
         // desiredAngle = __calculated_angle__;
         // desiredPower = __calculated_RPMS__;
+    }
+
+    private void calculateHoopChassis() {
+        hoop_rotation = 0.5 * (hoopr / 34.0);
+        //hoop_rotation = -pid.calculate(balla, 0);
+        if(hoopx < hoopRange[0]) {
+            hoop_forward_speed = hoopDistancePid.calculate(hoopx, hoopRange[0] + 3);
+        }
 
+        else if(hoopx > hoopRange[1]) {
+            hoop_forward_speed = hoopDistancePid.calculate(hoopx, hoopRange[1] - 3);
+        }
+
+        else {
+            hoop_forward_speed = 0D;
+        }
+
+        if(Math.abs(hoopr) <= hoopChassisThreshold) {
+            hoop_rotation = 0D;
+        }
+    }
+
+    private void calculateBallChassis() {
+        // ball_forward_speed = __calculated_forward_speed__;
+        // ball_sidespeed = __calculated_side_speed__;
+        ball_rotation = 0.5 * (ballr / 34.0);
```

Comment:
Similarly, this method can return the calculated value. We don't need to store it.

Reply:
done